fix: Make response handling a tad more robust

magandrez opened the pull request:
iratebirds.app as seen in production has an issue due to an upstream change the behavior of a 3rd party API: Macaulay Library's public API does no longer provide consistent support for the number of items contained in the response.

This change addresses three things:

- Updates Node version to latest version of Node 16 LTS.
- Updates dependencies (`npm upgrade`) and addresses dependencies' vulnerabilities registered by the Node tooling (`npm audit fix`), as a result, updating several dependencies, as seen in 1ec1375aad586b0516cc6c2adb914db06d552b6c.
- Uses length of response array to select an image to display, instead of relying on responses's `count` element, which seems to be broken in above mentioned public API.

magandrez commented:
As seen from my development environment upon launching the server (i.e.: `npm run serve`):

![image](https://user-images.githubusercontent.com/743549/206209603-e35c4649-1641-4df2-b55f-7b7f81e137c4.png)

wkmor1 commented:
Thanks for the help @magandrez! Even with your fixes and after I am upgraded the build image it is not building remotely on netlify. The deployment log (see below) doenst mean much to me. Any ideas? 

```
5:30:42 AM: build-image version: 4c0c1cadee6a31c9bb8d824514030009c4c05c6a (focal)
5:30:42 AM: build-image tag: v4.15.0
5:30:42 AM: buildbot version: cc0c986a947e274d444f48f63f797a5d01f2a89a
5:30:42 AM: Building without cache
5:30:42 AM: Starting to prepare the repo for build
5:30:42 AM: No cached dependencies found. Cloning fresh repo
5:30:42 AM: git clone https://github.com/luomus/iratebirds
5:30:43 AM: Preparing Git Reference pull/50/head
5:30:43 AM: Parsing package.json dependencies
5:30:44 AM: Section completed: initializing
5:30:44 AM: Starting build script
5:30:44 AM: Installing dependencies
5:30:44 AM: Python version set to 2.7
5:30:44 AM: Attempting node version 'v16.18.1' from .nvmrc
5:30:45 AM: v16.18.1 is already installed.
5:30:45 AM: Now using node v16.18.1 (npm v8.19.2)
5:30:45 AM: Enabling node corepack
5:30:45 AM: Started restoring cached build plugins
5:30:45 AM: Finished restoring cached build plugins
5:30:45 AM: Attempting ruby version 2.6.2, read from environment
5:30:46 AM: Required ruby-2.6.2 is not installed - installing.
5:30:46 AM: Searching for binary rubies, this might take some time.
5:30:46 AM: Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/ruby-2.6.2.tar.bz2
5:30:46 AM: Checking requirements for ubuntu.
5:30:46 AM: Requirements installation successful.
5:30:46 AM: ruby-2.6.2 - #configure
5:30:46 AM: ruby-2.6.2 - #download
5:30:46 AM: ruby-2.6.2 - #validate archive
5:30:50 AM: ruby-2.6.2 - #extract
5:30:51 AM: ruby-2.6.2 - #validate binary
5:30:51 AM: ruby-2.6.2 - #setup
5:30:52 AM: ruby-2.6.2 - #gemset created /opt/buildhome/.rvm/gems/ruby-2.6.2@global
5:30:52 AM: ruby-2.6.2 - #importing gemset /opt/buildhome/.rvm/gemsets/global.gems........................................
5:30:52 AM: ruby-2.6.2 - #generating global wrappers........
5:30:52 AM: ruby-2.6.2 - #gemset created /opt/buildhome/.rvm/gems/ruby-2.6.2
5:30:52 AM: ruby-2.6.2 - #importing gemsetfile /opt/buildhome/.rvm/gemsets/default.gems evaluated to empty gem list
5:30:53 AM: ruby-2.6.2 - #generating default wrappers........
5:30:53 AM: Using /opt/buildhome/.rvm/gems/ruby-2.6.2
5:30:53 AM: Using ruby version 2.6.2
5:30:53 AM: Using PHP version 8.0
5:30:53 AM: No npm workspaces detected
5:30:53 AM: Started restoring cached node modules
5:30:53 AM: Finished restoring cached node modules
5:30:53 AM: Installing NPM modules using NPM version 8.19.2
5:30:57 AM: npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
5:30:57 AM: npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
5:30:57 AM: npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
5:30:57 AM: npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
5:30:57 AM: npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
5:30:58 AM: npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
5:30:58 AM: npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
5:30:58 AM: npm WARN deprecated har-validator@5.1.5: this library is no longer supported
5:30:58 AM: npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
5:30:59 AM: npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin
5:30:59 AM: npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
5:31:00 AM: npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
5:31:00 AM: npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
5:31:00 AM: npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
5:31:00 AM: npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
5:31:00 AM: npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
5:31:00 AM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
5:31:00 AM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
5:31:01 AM: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
5:31:01 AM: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
5:31:06 AM: added 1512 packages, and audited 1513 packages in 13s
5:31:06 AM: 116 packages are looking for funding
5:31:06 AM:   run `npm fund` for details
5:31:06 AM: 42 vulnerabilities (1 low, 11 moderate, 20 high, 10 critical)
5:31:06 AM: To address issues that do not require attention, run:
5:31:06 AM:   npm audit fix
5:31:06 AM: To address all issues (including breaking changes), run:
5:31:06 AM:   npm audit fix --force
5:31:06 AM: Run `npm audit` for details.
5:31:07 AM: NPM modules installed
5:31:07 AM: Creating package sha
5:31:07 AM: Started restoring cached go cache
5:31:07 AM: Finished restoring cached go cache
5:31:07 AM: Installing Go version 1.12 (requested 1.12)
5:31:12 AM: unset GOOS;
5:31:12 AM: unset GOARCH;
5:31:12 AM: export GOROOT='/opt/buildhome/.gimme/versions/go1.12.linux.amd64';
5:31:12 AM: export PATH="/opt/buildhome/.gimme/versions/go1.12.linux.amd64/bin:${PATH}";
5:31:12 AM: go version >&2;
5:31:12 AM: export GIMME_ENV="/opt/buildhome/.gimme/env/go1.12.linux.amd64.env"
5:31:12 AM: go version go1.12 linux/amd64
5:31:12 AM: Detected 1 framework(s)
5:31:12 AM: "vue" at version "3.12.1"
5:31:12 AM: Installing missing commands
5:31:12 AM: Verify run directory
5:31:14 AM: ​
5:31:14 AM: ────────────────────────────────────────────────────────────────
5:31:14 AM:   Netlify Build                                                 
5:31:14 AM: ────────────────────────────────────────────────────────────────
5:31:14 AM: ​
5:31:14 AM: ❯ Version
5:31:14 AM:   @netlify/build 29.0.1
5:31:14 AM: ​
5:31:14 AM: ❯ Flags
5:31:14 AM:   baseRelDir: true
5:31:14 AM:   buildId: 63915a5fa8e81d2157b25ebb
5:31:14 AM:   deployId: 63915a5fa8e81d2157b25ebd
5:31:14 AM: ​
5:31:14 AM: ❯ Current directory
5:31:14 AM:   /opt/build/repo
5:31:14 AM: ​
5:31:14 AM: ❯ Config file
5:31:14 AM:   No config file was defined: using default values.
5:31:14 AM: ​
5:31:14 AM: ❯ Context
5:31:14 AM:   deploy-preview
5:31:14 AM: ​
5:31:14 AM: ────────────────────────────────────────────────────────────────
5:31:14 AM:   1. Build command from Netlify app                             
5:31:14 AM: ────────────────────────────────────────────────────────────────
5:31:14 AM: ​
5:31:14 AM: $ npm run build
5:31:14 AM: > iratebirds@0.1.0 build
5:31:14 AM: > vue-cli-service build
5:31:15 AM: -  Building for production...
5:31:15 AM: Starting type checking service...
5:31:15 AM: Using 1 worker with 2048MB memory limit
5:31:22 AM:  ERROR  Failed to compile with 1 error3:31:22 AM
5:31:22 AM:  error  in [internal]
5:31:22 AM: INTERNAL ERROR(undefined,undefined) Cannot destructure property 'script' of 'result.descriptor' as it is undefined.
5:31:22 AM: stack trace:
5:31:22 AM: TypeError: Cannot destructure property 'script' of 'result.descriptor' as it is undefined.
5:31:22 AM:     at Object.parseComponent (/opt/build/repo/node_modules/@vue/cli-plugin-typescript/vue-compiler-sfc-shim.js:6:13)
5:31:22 AM:     at Function.resolveScriptBlock (/opt/build/repo/node_modules/fork-ts-checker-webpack-plugin/lib/VueProgram.js:196:35)
5:31:22 AM:     at Object.host.getSourceFile (/opt/build/repo/node_modules/fork-ts-checker-webpack-plugin/lib/VueProgram.js:106:45)
5:31:22 AM:     at findSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101559:29)
5:31:22 AM:     at processImportedModules (/opt/build/repo/node_modules/typescript/lib/typescript.js:101900:25)
5:31:22 AM:     at findSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101607:17)
5:31:22 AM:     at /opt/build/repo/node_modules/typescript/lib/typescript.js:101436:85
5:31:22 AM:     at getSourceFileFromReferenceWorker (/opt/build/repo/node_modules/typescript/lib/typescript.js:101403:34)
5:31:22 AM:     at processSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101436:13)
5:31:22 AM:     at processRootFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101259:13)
5:31:22 AM:  ERROR  Build failed with errors.
5:31:22 AM: ​
5:31:22 AM: ────────────────────────────────────────────────────────────────
5:31:22 AM:   "build.command" failed                                        
5:31:22 AM: ────────────────────────────────────────────────────────────────
5:31:22 AM: ​
5:31:22 AM:   Error message
5:31:22 AM:   Command failed with exit code 1: npm run build (https://ntl.fyi/exit-code-1)
5:31:22 AM: ​
5:31:22 AM:   Error location
5:31:22 AM:   In Build command from Netlify app:
5:31:22 AM:   npm run build
5:31:22 AM: ​
5:31:22 AM:   Resolved config
5:31:22 AM:   build:
5:31:22 AM:     command: npm run build
5:31:22 AM:     commandOrigin: ui
5:31:22 AM:     environment:
5:31:22 AM:       - REVIEW_ID
5:31:22 AM:     publish: /opt/build/repo/dist
5:31:22 AM:     publishOrigin: ui
5:31:22 AM: Caching artifacts
5:31:22 AM: Started saving node modules
5:31:22 AM: Finished saving node modules
5:31:22 AM: Started saving build plugins
5:31:22 AM: Finished saving build plugins
5:31:22 AM: Started saving pip cache
5:31:23 AM: Finished saving pip cache
5:31:23 AM: Started saving emacs cask dependencies
5:31:24 AM: Creating deploy upload records
5:31:23 AM: Finished saving emacs cask dependencies
5:31:23 AM: Started saving maven dependencies
5:31:23 AM: Finished saving maven dependencies
5:31:23 AM: Started saving boot dependencies
5:31:23 AM: Finished saving boot dependencies
5:31:24 AM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
5:31:23 AM: Started saving rust rustup cache
5:31:23 AM: Finished saving rust rustup cache
5:31:23 AM: Started saving go dependencies
5:31:23 AM: Finished saving go dependencies
5:31:24 AM: Cached ruby version 2.6.2
5:31:24 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
5:31:24 AM: Failing build: Failed to build site
5:31:24 AM: Finished processing build request in 42.32311663s
```

magandrez commented:
> Thanks for the help @magandrez! Even with your fixes and after I am upgraded the build image it is not building remotely on netlify. The deployment log (see below) doenst mean much to me. Any ideas? 
> 
> 
> 
> ```
> 
> 5:30:42 AM: build-image version: 4c0c1cadee6a31c9bb8d824514030009c4c05c6a (focal)
> 
> 5:30:42 AM: build-image tag: v4.15.0
> 
> 5:30:42 AM: buildbot version: cc0c986a947e274d444f48f63f797a5d01f2a89a
> 
> 5:30:42 AM: Building without cache
> 
> 5:30:42 AM: Starting to prepare the repo for build
> 
> 5:30:42 AM: No cached dependencies found. Cloning fresh repo
> 
> 5:30:42 AM: git clone https://github.com/luomus/iratebirds
> 
> 5:30:43 AM: Preparing Git Reference pull/50/head
> 
> 5:30:43 AM: Parsing package.json dependencies
> 
> 5:30:44 AM: Section completed: initializing
> 
> 5:30:44 AM: Starting build script
> 
> 5:30:44 AM: Installing dependencies
> 
> 5:30:44 AM: Python version set to 2.7
> 
> 5:30:44 AM: Attempting node version 'v16.18.1' from .nvmrc
> 
> 5:30:45 AM: v16.18.1 is already installed.
> 
> 5:30:45 AM: Now using node v16.18.1 (npm v8.19.2)
> 
> 5:30:45 AM: Enabling node corepack
> 
> 5:30:45 AM: Started restoring cached build plugins
> 
> 5:30:45 AM: Finished restoring cached build plugins
> 
> 5:30:45 AM: Attempting ruby version 2.6.2, read from environment
> 
> 5:30:46 AM: Required ruby-2.6.2 is not installed - installing.
> 
> 5:30:46 AM: Searching for binary rubies, this might take some time.
> 
> 5:30:46 AM: Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/ruby-2.6.2.tar.bz2
> 
> 5:30:46 AM: Checking requirements for ubuntu.
> 
> 5:30:46 AM: Requirements installation successful.
> 
> 5:30:46 AM: ruby-2.6.2 - #configure
> 
> 5:30:46 AM: ruby-2.6.2 - #download
> 
> 5:30:46 AM: ruby-2.6.2 - #validate archive
> 
> 5:30:50 AM: ruby-2.6.2 - #extract
> 
> 5:30:51 AM: ruby-2.6.2 - #validate binary
> 
> 5:30:51 AM: ruby-2.6.2 - #setup
> 
> 5:30:52 AM: ruby-2.6.2 - #gemset created /opt/buildhome/.rvm/gems/ruby-2.6.2@global
> 
> 5:30:52 AM: ruby-2.6.2 - #importing gemset /opt/buildhome/.rvm/gemsets/global.gems........................................
> 
> 5:30:52 AM: ruby-2.6.2 - #generating global wrappers........
> 
> 5:30:52 AM: ruby-2.6.2 - #gemset created /opt/buildhome/.rvm/gems/ruby-2.6.2
> 
> 5:30:52 AM: ruby-2.6.2 - #importing gemsetfile /opt/buildhome/.rvm/gemsets/default.gems evaluated to empty gem list
> 
> 5:30:53 AM: ruby-2.6.2 - #generating default wrappers........
> 
> 5:30:53 AM: Using /opt/buildhome/.rvm/gems/ruby-2.6.2
> 
> 5:30:53 AM: Using ruby version 2.6.2
> 
> 5:30:53 AM: Using PHP version 8.0
> 
> 5:30:53 AM: No npm workspaces detected
> 
> 5:30:53 AM: Started restoring cached node modules
> 
> 5:30:53 AM: Finished restoring cached node modules
> 
> 5:30:53 AM: Installing NPM modules using NPM version 8.19.2
> 
> 5:30:57 AM: npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
> 
> 5:30:57 AM: npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
> 
> 5:30:57 AM: npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> 
> 5:30:57 AM: npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
> 
> 5:30:57 AM: npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
> 
> 5:30:58 AM: npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
> 
> 5:30:58 AM: npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
> 
> 5:30:58 AM: npm WARN deprecated har-validator@5.1.5: this library is no longer supported
> 
> 5:30:58 AM: npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
> 
> 5:30:59 AM: npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin
> 
> 5:30:59 AM: npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
> 
> 5:31:00 AM: npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
> 
> 5:31:00 AM: npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
> 
> 5:31:00 AM: npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
> 
> 5:31:00 AM: npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
> 
> 5:31:00 AM: npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
> 
> 5:31:00 AM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
> 
> 5:31:00 AM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
> 
> 5:31:01 AM: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> 
> 5:31:01 AM: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> 
> 5:31:06 AM: added 1512 packages, and audited 1513 packages in 13s
> 
> 5:31:06 AM: 116 packages are looking for funding
> 
> 5:31:06 AM:   run `npm fund` for details
> 
> 5:31:06 AM: 42 vulnerabilities (1 low, 11 moderate, 20 high, 10 critical)
> 
> 5:31:06 AM: To address issues that do not require attention, run:
> 
> 5:31:06 AM:   npm audit fix
> 
> 5:31:06 AM: To address all issues (including breaking changes), run:
> 
> 5:31:06 AM:   npm audit fix --force
> 
> 5:31:06 AM: Run `npm audit` for details.
> 
> 5:31:07 AM: NPM modules installed
> 
> 5:31:07 AM: Creating package sha
> 
> 5:31:07 AM: Started restoring cached go cache
> 
> 5:31:07 AM: Finished restoring cached go cache
> 
> 5:31:07 AM: Installing Go version 1.12 (requested 1.12)
> 
> 5:31:12 AM: unset GOOS;
> 
> 5:31:12 AM: unset GOARCH;
> 
> 5:31:12 AM: export GOROOT='/opt/buildhome/.gimme/versions/go1.12.linux.amd64';
> 
> 5:31:12 AM: export PATH="/opt/buildhome/.gimme/versions/go1.12.linux.amd64/bin:${PATH}";
> 
> 5:31:12 AM: go version >&2;
> 
> 5:31:12 AM: export GIMME_ENV="/opt/buildhome/.gimme/env/go1.12.linux.amd64.env"
> 
> 5:31:12 AM: go version go1.12 linux/amd64
> 
> 5:31:12 AM: Detected 1 framework(s)
> 
> 5:31:12 AM: "vue" at version "3.12.1"
> 
> 5:31:12 AM: Installing missing commands
> 
> 5:31:12 AM: Verify run directory
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:14 AM:   Netlify Build                                                 
> 
> 5:31:14 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ❯ Version
> 
> 5:31:14 AM:   @netlify/build 29.0.1
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ❯ Flags
> 
> 5:31:14 AM:   baseRelDir: true
> 
> 5:31:14 AM:   buildId: 63915a5fa8e81d2157b25ebb
> 
> 5:31:14 AM:   deployId: 63915a5fa8e81d2157b25ebd
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ❯ Current directory
> 
> 5:31:14 AM:   /opt/build/repo
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ❯ Config file
> 
> 5:31:14 AM:   No config file was defined: using default values.
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ❯ Context
> 
> 5:31:14 AM:   deploy-preview
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:14 AM:   1. Build command from Netlify app                             
> 
> 5:31:14 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:14 AM: ​
> 
> 5:31:14 AM: $ npm run build
> 
> 5:31:14 AM: > iratebirds@0.1.0 build
> 
> 5:31:14 AM: > vue-cli-service build
> 
> 5:31:15 AM: -  Building for production...
> 
> 5:31:15 AM: Starting type checking service...
> 
> 5:31:15 AM: Using 1 worker with 2048MB memory limit
> 
> 5:31:22 AM:  ERROR  Failed to compile with 1 error3:31:22 AM
> 
> 5:31:22 AM:  error  in [internal]
> 
> 5:31:22 AM: INTERNAL ERROR(undefined,undefined) Cannot destructure property 'script' of 'result.descriptor' as it is undefined.
> 
> 5:31:22 AM: stack trace:
> 
> 5:31:22 AM: TypeError: Cannot destructure property 'script' of 'result.descriptor' as it is undefined.
> 
> 5:31:22 AM:     at Object.parseComponent (/opt/build/repo/node_modules/@vue/cli-plugin-typescript/vue-compiler-sfc-shim.js:6:13)
> 
> 5:31:22 AM:     at Function.resolveScriptBlock (/opt/build/repo/node_modules/fork-ts-checker-webpack-plugin/lib/VueProgram.js:196:35)
> 
> 5:31:22 AM:     at Object.host.getSourceFile (/opt/build/repo/node_modules/fork-ts-checker-webpack-plugin/lib/VueProgram.js:106:45)
> 
> 5:31:22 AM:     at findSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101559:29)
> 
> 5:31:22 AM:     at processImportedModules (/opt/build/repo/node_modules/typescript/lib/typescript.js:101900:25)
> 
> 5:31:22 AM:     at findSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101607:17)
> 
> 5:31:22 AM:     at /opt/build/repo/node_modules/typescript/lib/typescript.js:101436:85
> 
> 5:31:22 AM:     at getSourceFileFromReferenceWorker (/opt/build/repo/node_modules/typescript/lib/typescript.js:101403:34)
> 
> 5:31:22 AM:     at processSourceFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101436:13)
> 
> 5:31:22 AM:     at processRootFile (/opt/build/repo/node_modules/typescript/lib/typescript.js:101259:13)
> 
> 5:31:22 AM:  ERROR  Build failed with errors.
> 
> 5:31:22 AM: ​
> 
> 5:31:22 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:22 AM:   "build.command" failed                                        
> 
> 5:31:22 AM: ────────────────────────────────────────────────────────────────
> 
> 5:31:22 AM: ​
> 
> 5:31:22 AM:   Error message
> 
> 5:31:22 AM:   Command failed with exit code 1: npm run build (https://ntl.fyi/exit-code-1)
> 
> 5:31:22 AM: ​
> 
> 5:31:22 AM:   Error location
> 
> 5:31:22 AM:   In Build command from Netlify app:
> 
> 5:31:22 AM:   npm run build
> 
> 5:31:22 AM: ​
> 
> 5:31:22 AM:   Resolved config
> 
> 5:31:22 AM:   build:
> 
> 5:31:22 AM:     command: npm run build
> 
> 5:31:22 AM:     commandOrigin: ui
> 
> 5:31:22 AM:     environment:
> 
> 5:31:22 AM:       - REVIEW_ID
> 
> 5:31:22 AM:     publish: /opt/build/repo/dist
> 
> 5:31:22 AM:     publishOrigin: ui
> 
> 5:31:22 AM: Caching artifacts
> 
> 5:31:22 AM: Started saving node modules
> 
> 5:31:22 AM: Finished saving node modules
> 
> 5:31:22 AM: Started saving build plugins
> 
> 5:31:22 AM: Finished saving build plugins
> 
> 5:31:22 AM: Started saving pip cache
> 
> 5:31:23 AM: Finished saving pip cache
> 
> 5:31:23 AM: Started saving emacs cask dependencies
> 
> 5:31:24 AM: Creating deploy upload records
> 
> 5:31:23 AM: Finished saving emacs cask dependencies
> 
> 5:31:23 AM: Started saving maven dependencies
> 
> 5:31:23 AM: Finished saving maven dependencies
> 
> 5:31:23 AM: Started saving boot dependencies
> 
> 5:31:23 AM: Finished saving boot dependencies
> 
> 5:31:24 AM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
> 
> 5:31:23 AM: Started saving rust rustup cache
> 
> 5:31:23 AM: Finished saving rust rustup cache
> 
> 5:31:23 AM: Started saving go dependencies
> 
> 5:31:23 AM: Finished saving go dependencies
> 
> 5:31:24 AM: Cached ruby version 2.6.2
> 
> 5:31:24 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
> 
> 5:31:24 AM: Failing build: Failed to build site
> 
> 5:31:24 AM: Finished processing build request in 42.32311663s
> 
> ```

Oh I see the problem

I'll try to find some time to fix it between today and tomorrow 👍🏻

wkmor1 commented:
Awesome! Merging..
